js/src/mesh.ts

```diff
@@ -341,26 +353,35 @@ class MeshView extends widgets.WidgetView {
         if (this.model.get("line_material")) {
             this.line_material_rgb.copy(this.model.get("line_material").obj);
         }
-        this.material.defines = {...this.scale_defines};
-        this.material.defines.USE_COLORMAP = this.model.get("color_scale") !== null;
-        this.material_rgb.defines = {USE_RGB: true, ...this.scale_defines};
-        this.line_material.defines = {AS_LINE: true, ...this.scale_defines};
-        this.line_material_rgb.defines = {AS_LINE: true, USE_RGB: true, ...this.scale_defines};
+        this.lighting_model = materialToLightingModel(this.material);
+        // update material defines in order to run correct shader code
+        this.material.defines = {USE_COLOR: true,  USE_COLORMAP: this.model.get("color_scale") !== null, ...this.scale_defines};
+        this.material.defines[`AS_${this.lighting_model}`] = true;
         this.material.extensions = {derivatives: true};
+        this.material_rgb.defines = {AS_COORDINATE: true, USE_COLOR: true, ...this.scale_defines};
+        this.line_material.defines = {IS_LINE: true, ...this.scale_defines};
+        this.line_material.defines[`AS_${this.lighting_model}`] = true;
+        this.line_material_rgb.defines = {IS_LINE: true, AS_COORDINATE: true, USE_COLOR: true, ...this.scale_defines};
+
         // locally and the visible with this object's visible trait
         this.material.visible = this.material.visible && this.model.get("visible");
         this.material_rgb.visible = this.material.visible && this.model.get("visible");
         this.line_material.visible = this.line_material.visible && this.model.get("visible");
         this.line_material_rgb.visible = this.line_material.visible && this.model.get("visible");
+
         this.materials.forEach((material) => {
-            material.vertexShader = (require("raw-loader!../glsl/mesh-vertex.glsl") as any).default;
-            material.fragmentShader = (require("raw-loader!../glsl/mesh-fragment.glsl") as any).default;
-            material.uniforms = this.uniforms;
-            material.depthWrite = true;
-            material.transparant = true;
-            material.depthTest = true;
-            patchMaterial(material);
+            material.onBeforeCompile = (shader) => {
+                shader.vertexShader = vertexShader;
+                shader.fragmentShader = fragmentShader;
+                shader.uniforms = {...shader.uniforms, ...this.uniforms};
+                patchShader(shader);
+            };
+            material.needsUpdate = true;
+            material.lights = true;
```

Comment:
```suggestion
```
No need for this if we set it in the ShaderMaterial constructor from the Python side (Phong, Physical etc don't seem to need this)